### PR TITLE
add multiple storage backend options and related instructions for docker compose deployment

### LINKED ISSUE #953: Document configuring the docker compose cluster to use GCS
https://github.com/risingwavelabs/risingwave/issues/10169

### LINKED ISSUE #954: Document configuring the docker compose cluster to use Azure Blob
https://github.com/risingwavelabs/risingwave/issues/10167

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

 Add S3, S3-compatible, GCS, AliCloud OSS, Azure Blob, and HDFS to the docker compose instructions.
 As there'll be some upcoming code changes to data_directory, this PR will be in Draft until these changes are merged.

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #953 and #954

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Review by AbdullahWins — COMMENTED** [review]
Nice Work!

File: docs/deploy/risingwave-trial.md [hunk 1]
```diff
@@ -255,13 +255,83 @@ RisingWave also incorporates these third-party components:
...
+In `/docker-compose-with-hdfs.yml`, specify the cluster name via the `hummock+hdfs` parameter.
+
+```bash
+- "hummock+hdfs://<cluster_name>"
+```
+
+Run the following command to start a RisingWave cluster:
+
+```shell
+`docker-compose -f docker-compose-with-hdfs.yml up
```
Comment (wcy-fdu) [inline docs/deploy/risingwave-trial.md]: typo

**Review by wcy-fdu — APPROVED** [review]
Rest LGTM! Thanks!

File: docs/deploy/risingwave-trial.md [hunk 2]
```diff
@@ -255,13 +255,83 @@ RisingWave also incorporates these third-party components:
...
+
+```shell
+docker compose up -d
+```
+
+#### Start a RisingWave cluster with S3 or S3-compatible storage
+
+To use S3 as the storage backend, configure your AWS credential information in `/docker/aws.env`.
+
+To use S3-compatible storage options like Tencent Cloud COS, you need to configure the endpoint via the `RW_S3_ENDPOINT` parameter in `/docker/aws.env`.
```
Comment (wcy-fdu) [inline docs/deploy/risingwave-trial.md]: Add a note here:
Do not contain bucket name in `RW_S3_ENDPOINT`.

**Pull request merged**